batchflow/models/eager_torch/layers/pooling.py

```diff
@@ -0,0 +1,131 @@
+""" Pooling layers. """
+#pylint: disable=not-callable, invalid-name
+import torch.nn as nn
+import torch.nn.functional as F
+
+from ..utils import get_shape, get_num_dims, calc_padding
+
+
+
+class BasePool(nn.Module):
+    """ A universal pooling layer. """
+    LAYER = None
+    LAYERS = {}
+
+    def __init__(self, inputs=None, pool_size=2, pool_strides=2, padding='same', **kwargs):
+        super().__init__()
+        self.padding = None
+
+        if self.LAYER is not None:
```

Comment:
I don't like this if-else block that contains most part of functionality. What about having 2 base classes - BasePool that selects nn classes depending on input size and, say, `PoolingOpSelector` that selects bf class depending on `op`